common/models/lessonPlanStep.ts

```diff
@@ -0,0 +1,46 @@
+import { Entity, PrimaryGeneratedColumn, Column, BaseEntity, ManyToMany, OneToOne, JoinTable, JoinColumn, OneToMany, ManyToOne } from 'typeorm'
+
+import { LessonPlanStepInterface } from '@interfaces/lessonPlanStep.interface'
+import { User } from '@models/user'
+import { Course } from '@models/course'
+import { Problem } from '@models/problem'
+
+@Entity()
+export class LessonPlanStep extends BaseEntity implements LessonPlanStepInterface {
+
+	@PrimaryGeneratedColumn("uuid")
+	public id: string
+
+	@Column()
+	public title?: string;
+
+	@Column()
+	public description?: string;
+
+	@Column()	
+	public estimatedDuration?: number;
+
+	@ManyToOne(type => Problem, { eager: true })
```

Comment:
whats eager?

Reply:
it makes it load in the relations automatically without having to do a populate/join on our end when querying.